File: backend/WatchNow/src/main/java/com/intiformation/WatchNow/service/UtilisateurLoginService.java
package com.intiformation.WatchNow.service;

import java.util.List;

import com.intiformation.WatchNow.model.UtilisateurLogin;

public interface UtilisateurLoginService
{
	public List<UtilisateurLogin> get();
    public UtilisateurLogin get(int id);
    public UtilisateurLogin save(UtilisateurLogin ul);
    public void delete(int id);
    public void update(UtilisateurLogin ul);
}

File: frontend/WatchNow/src/app/service/utilisateur-login.service.ts
import { environment } from './../../environments/environment';
import { Observable } from 'rxjs';
import { UtilisateurLogin } from './../model/utilisateur-login.model';
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class UtilisateurLoginService {

  constructor(private httpClient: HttpClient) { }

  getAll(): Observable<UtilisateurLogin[]> {
    return this.httpClient.get<UtilisateurLogin[]>(environment.url + "utilisateur_login/index")
  }

  getById(id: number): Observable<UtilisateurLogin> {
    return this.httpClient.get<UtilisateurLogin>(environment.url + "utilisateur_login/index/" + id)
  }

  save(utilisateurLogin: UtilisateurLogin): Observable<UtilisateurLogin> {
    return this.httpClient.post<UtilisateurLogin>(environment.url + "utilisateur_login/save", utilisateurLogin)
  }

  delete(id: number): Observable<void> {
    return this.httpClient.delete<void>(environment.url + "utilisateur_login/delete/" + id)
  }

  update(utilisateurLogin: UtilisateurLogin): Observable<UtilisateurLogin> {
    return this.httpClient.put<UtilisateurLogin>(environment.url + "utilisateur_login/update", utilisateurLogin)
  }
}
